src/vunnel/provider.py

```diff
@@ -136,6 +166,41 @@ def _update(self) -> None:
         else:
             self.logger.debug("skipping recording of workspace state (no new results found)")
 
+    def _fetch_or_use_results_archive(self) -> tuple[list[str], int]:
+        listing_doc = self._fetch_listing_document()
+        latest_entry = listing_doc.latest_entry(schema_version=self.version())
+        if not latest_entry:
+            raise RuntimeError("no listing entry found")
+
+        if self._has_newer_archive(latest_entry=latest_entry):
+            self._prep_workspace_from_listing_entry(entry=latest_entry)
+        state = self.workspace.state()
+        return state.urls, state.result_count(self.workspace.path)
+
+    def _fetch_listing_document(self) -> distribution.ListingDocument:
+        url = self.runtime_cfg.import_url(provider_name=self.name())
+        resp = http.get(url, logger=self.logger)
+        resp.raise_for_status()
+
+        return distribution.ListingDocument.from_dict(resp.json())
+
+    def _has_newer_archive(self, latest_entry: distribution.ListingEntry) -> bool:
+        # TODO: can do checksum comparison here
```

Comment:
@wagoodman I'm not sure what checksum comparison you mean here. Do you mean making sure the listing entry passes its own checksum validation and falling back to the next oldest if it doesnt?